var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -778,6 +793,10 @@ def define_tpl(trilinos_name, spack_name, have_dep):
         for tpl_name, dep_name in tpl_dep_map:
             define_tpl(tpl_name, dep_name, dep_name in spec)
 
+        # External Kokkos
+        if spec.satisfies("@14.4.0 +kokkos"):
```

Comment:
Sorry, I should be more clear in my question. Will the Spack build of Trilinos now only admit an external Kokkos? It seems like this logic will catch the `+kokkos` variant (like when you want Trilinos with Kokkos enabled) and then this will add a dependency on the external Kokkos. I was expecting a `^kokkos` to indicate an external dependency was requested.

As-is, it reads like the only way to not depend on Kokkos externally is to disabled the variant with `~kokkos`, however this will also disable Kokkos built in Trilinos (effectively removing the option for a vendored build of Kokkos in Trilinos when built by Spack).

Maybe the intent is to only offer an external Kokkos in Spack for 14.4 onward, which makes sense from the perspective of reuse of Kokkos by other packages in Spack.

Reply:
You are right, the way this PR is constructed, there is not a way to `spack install trilinos@14.4.0 +kokkos` with vendor'd kokkos, although it would be possible to use vendor'd kokkos with `<14.4.0`.

Without introducing a new variant like `kokkos-external` I don't know if we could allow the option of both vendor'd and external kokkos.

Reply:
Spack should use as few vendored components as possible.

Reply:
This seems reasonable and allows maximal reuse with other packages. Allowing for an internal Kokkos build in Spack, while the option also exists for an external one, would create a lot of complication as it could become possible to build Kokkos in Trilinos as well as outside of Trilinos, possibly with different versions and configurations.

Obviously this PR has no effect on Trilinos users who download and install outside of Spack, leaving the vendored Kokkos version as an option for those users. Thank you all for the clarification.